crates/bevy_reflect/src/path.rs

```diff
@@ -136,12 +141,48 @@ impl GetPath for dyn Reflect {
                                     index: current_index,
                                 })
                             }
+                        },
+                        Some(Token::SingleQuote) => {
```

Comment:
Also notably, the two match branches in `path()` and `path_mut()` are exactly the same (minus the `get`/`get_mut` call on the map. It's probably worth DRY'ing this out into a similar function to reduce code duplication.

Reply:
I moved the code to find the complete quoted identifier to its own function, which only leaves the `match reflect_ref` vs `match reflect_mut` part in those branches. Not sure if it possible/feasible to further reduce duplication here without creating some kind ob abstraction over mutability?